samples/sample_encode/src/pipeline_encode.cpp

```diff
@@ -463,8 +463,15 @@ mfxStatus CEncodingPipeline::InitMfxEncParams(sInputParams *pInParams)
 
     m_mfxEncParams.mfx.NumSlice = pInParams->nNumSlice;
     ConvertFrameRate(pInParams->dFrameRate, &m_mfxEncParams.mfx.FrameInfo.FrameRateExtN, &m_mfxEncParams.mfx.FrameInfo.FrameRateExtD);
-    m_mfxEncParams.mfx.EncodedOrder            = 0; // binary flag, 0 signals encoder to take frames in display order
 
+    m_bQPFileMode = pInParams->QPFileMode ? 1 : 0;
+    m_mfxEncParams.mfx.EncodedOrder = m_bQPFileMode; // binary flag, 0 signals encoder to take frames in display order
```

Comment:
and this is ```bool``` to ```short unsigned int``` implicit conversion, it will work fine, but you may convert to integer first as you did on line 467

Reply:
changed